### ISSUE TITLE
A Day in the Life of a Windows Sysadmin

### ISSUE THREAD

**Issue by wil-ch** [wil-ch/Basic-network-design#1]
I have been working on a project titled "A Day in the Life of a Windows Sysadmin," and I'd like to take this opportunity to present it to you. This project focuses on enhancing the security and management of Windows systems by creating and implementing Group Policy Objects (GPOs) and PowerShell scripts. Here's an overview of the project's key tasks:

Task 1: Create a GPO - Disable Local Link Multicast Name Resolution (LLMNR) In this task, I investigated and mitigated a potential security vulnerability in Windows domains related to Local Link Multicast Name Resolution (LLMNR). LLMNR is a protocol used as a backup for DNS in Windows, and it can pose a security risk if not properly configured. I created a GPO called "No LLMNR" to disable LLMNR on domain-joined Windows machines, thereby preventing them from trusting potentially malicious location responses from attackers.

![Image](https://github.com/Cwil99/Basic-network-design/assets/87434592/a0885060-20ff-4a7a-bd71-5d54b5a0a006)

Task 2: Create a GPO - Account Lockout For this task, I implemented an account lockout policy to enhance security and compliance on Windows workstations. An account lockout policy helps defend against brute-force attacks by temporarily disabling accounts after a certain number of failed login attempts. I named the GPO "Account Lockout" and applied a reasonable lockout policy to the Windows 10 machine within the GC Computers OU.

![Image](https://github.com/Cwil99/Basic-network-design/assets/87434592/2fb7b857-ec65-426d-a4ef-1b71feeee458)

Task 3: Create a GPO - Enabling Verbose PowerShell Logging and Transcription To address the growing concern of PowerShell usage in cyberattacks, I created a GPO called "PowerShell Logging" to enable enhanced PowerShell logging and transcription. This GPO combined multiple policies, including enabling module logging, script block logging, script execution, and PowerShell transcription. These measures enhance visibility and combat obfuscated PowerShell payloads, contributing to improved security.

![Image](https://github.com/Cwil99/Basic-network-design/assets/87434592/987e0b55-69d1-401c-9955-9c206bd5c11f)

Task 4: Create a Script - Enumerate Access Control Lists In this task, I developed a PowerShell script called "enum_acls.ps1" to enumerate the Access Control Lists (ACLs) of files and subdirectories within the current working directory. The script utilizes a foreach loop to iterate through the directory contents, retrieves the ACLs using the Get-Acl cmdlet, and displays the results. This script aids in assessing and managing file and directory permissions.

```
$directory = Get-ChildItem .\
foreach ($item in $directory) {
  Get-Acl $item
Write-output $directory.name done !
}
```

Task 5: Verify Your PowerShell Logging GPO To ensure the effectiveness of the PowerShell logging GPO, I tested it by running a "gpupdate," followed by executing the "enum_acls.ps1" script. This process allowed me to verify that the GPO successfully created verbose PowerShell logs in the designated directory.

![Image](https://github.com/Cwil99/Basic-network-design/assets/87434592/b8f39e22-ed19-4f05-9751-fd00aa087604)

In summary, "A Day in the Life of a Windows Sysadmin" is a project that focuses on enhancing the security and management of Windows systems through the implementation of GPOs and PowerShell scripts. By addressing vulnerabilities, improving security policies, and enabling comprehensive logging, the project contributes to a more secure and well-managed Windows environment. Thank you for considering my project, and I'm open to discussing any further details or questions you may have.
